hydration/vectors.py

```diff
@@ -172,6 +175,17 @@ def value(self, value):
         # This assumes that the Struct will update the length field's value
         self.length = len(value)
 
+    def from_stream(self, read_func: Callable[[int], bytes]):
+        if isinstance(self.type, Field):
+            return super().from_bytes(read_func(len(self) * len(self.type)))
+        else:
+            val = []
+            for _ in range(len(self)):
+                next_obj = self.type.from_stream(read_func)
+                val.append(next_obj)
+            self.value = val
+            return self
+
     def from_bytes(self, data: bytes):
```

Comment:
`from_bytes` implementations can be deleted once it calls from_stream by default.